File: cs50-practice/week6/mario.py
# ask user for height
while True:
    try:
        height = int(input("Height: "))
        if height > 0 and height < 9:
            break
    except ValueError:
        pass

# print column
for i in range(height):
    # print spaces for left pyramid
    for j in range(height - i - 1):
        print(" ", end="")
    # print bricks for left pyramid
    for k in range(i + 1):
        print("#", end="")
    # print two spaces to separate
    print("  ", end="")
    # print bricks for right pyramid
    print("#" * (i + 1))

File: cs50-practice/week1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Get user input for height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    // Keeps asking for height until it is between 1-8
    while (height < 1 || height > 8);
    // Print pyramid height
    for (int i = 0; i < height; i++)
    {
        // Print spaces
        for (int j = 0; j < (height - i - 1); j++)
        {
            printf(" ");
        }
        // Print bricks
        for (int k = 0; k < (i + 1); k++)
        {
            printf("#");
        }
        // Print two spaces
        printf("  ");
        // Print right hashes
        for (int m = 0; m < (i + 1); m++)
        {
            printf("#");
        }
        // Print new line.
        printf("\n");
    }
}
